src/uwuw/uwuw_preprocessor.cpp

```diff
@@ -0,0 +1,545 @@
+  std::string entity_type;
+  if (dimension == 3) {
+    rval = DAG->measure_volume(entity, size);
+    entity_type = "Volume";
+  }
+  if (dimension == 2) {
+    rval = DAG->measure_area(entity, size);
+    entity_type = "Surface";
+  }
+
+  std::cout << entity_type << std::endl;
+
+  // to convert the int to a string
+  std::stringstream ss;
+  ss << entity_id;
+  std::string string_id = ss.str();
+
+  // make the tally name
+  // tally name is first two chars or particle name
+  // first 2 chars of tally type
+  // plus the entity id
+  std::string tally_name = "";
+  if (entity_id < 1000)
+    tally_name = particle.substr(0, 2) + tally_type.substr(0, 2)
+                 + std::string(string_id);
+  if (entity_id >= 1000 && entity_id < 10000)
+    tally_name = particle.substr(0, 2) + tally_type.substr(0, 1)
+                 + std::string(string_id);
+  if (entity_id >= 10000 && entity_id < 100000)
+    tally_name = particle.substr(0, 1) + tally_type.substr(0, 1)
+                 + std::string(string_id);
+
+  // NB We rely on the code writing out the tally name to deal with allignment
+
+  std::transform(tally_name.begin(), tally_name.end(), tally_name.begin(), toupper);
+
+  // keep list of particles and tally types
+  tally_info tally_data;
+
+  // set the structure data for the return
+  tally_data.particle_name = particle;
+  tally_data.tally_type = tally_type;
+  tally_data.tally_name = tally_name;
+  tally_data.dimension = dimension;
+  tally_data.entity_type = entity_type;
+  tally_data.entity_size = size;
+  tally_data.entity_id = entity_id;
+
+  return tally_data;
+}
+
+void uwuw_preprocessor::check_tally_props(std::string particle_name,
+                                          std::string tally_type,
+                                          int dimension,
+                                          int entityid) {
+
+  // check to make sure the particle is allowed
+  std::cout << "particle_name " << particle_name << std::endl;
+  std::cout << "valid " << pyne::particle::is_valid(particle_name) << std::endl;
+  if (!pyne::particle::is_valid(particle_name)) {
+    if (dimension == 3) {
+      std::cout << "Error: In tally on cell, ";
+    } else if (dimension == 2) {
+      std::cout << "Error: In tally on surface, ";
+    }
+    std::cout << particle_name << std::endl;
+    std::cout << entityid << " , particle " << particle_name;
+    std::cout << " not a valid particle type" << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  // check to make sure the particle is allowed
+  if ((tally_type != "Flux") && (tally_type != "Current")) {
+    if (dimension == 3) {
+      std::cout << "Error: In tally on cell, ";
+    } else if (dimension == 2) {
+      std::cout << "Error: In tally on surface, ";
+    }
+    std::cout << entityid << " , tally type " << tally_type;
+    std::cout << " not a valid tally" << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  return;
+}
+
+
+// constructor
+name_concatenator::name_concatenator() {
```

Comment:
Depending on optimization level, it could be better to have these empty bodies in the header file so that when objects are created/destroyed it is known that no additional code needs to be called out to.